dev/dev-env-setup.sh

```diff
@@ -171,8 +171,11 @@ version_gt() {
 
 # Check if pyenv is installed and offer to install it if not present
 check_and_install_pyenv() {
-  pyenv_exist=$(command -v pyenv)
-
+  if [ -z "$GITHUB_ACTIONS" ]; then
```

Comment:
let's do the apt-get install for python3-env - should be 3-6 minutes of additional runtime

Reply:
I think this is a better approach https://github.com/mlflow/mlflow/pull/12485/files#r1655687590